pwndbg/remote.py

```diff
@@ -11,23 +11,9 @@
 
 import gdb
 
+import pwndbg.memoize
 
-def is_remote():
-    # N.B.: We cannot use "info program" because of:
-    # https://sourceware.org/bugzilla/show_bug.cgi?id=18335
-    #
-    # return 'serial line' in gdb.execute('info program',to_string=True,)
-    info_file = gdb.execute('info file',to_string=True,from_tty=False)
-
-    # target remote
-    if 'Remote serial target' in info_file:
-        return True
-
-    # target extended-remote
-    if 'Extended remote serial target' in info_file:
-        return True
 
-    if 'Debugging a target over a serial line.' in info_file:
-        return True
-
-    return False
+@pwndbg.memoize.reset_on_exit
```

Comment:
Is `reset_on_exit` correct here? I.e. if we connect to different target, this should be resetted too.

Reply:
I did the following test:
* Add a `print` in `is_remote` so we can know when it's called
* `gdb bash` and I notice `is_remote` is called
* In this gdb session, run `target remote 127.0.0.1:1234` to connect to a gdbserver, and I notice `is_remote` is called again

So I think `reset_on_exit` will also be invoked when we connect to a different target, we don't need to change anything.